leech/src/modules/dns/errors.rs

```diff
@@ -0,0 +1,21 @@
+//! Holds all the errors from dns resolution
+
+use std::fmt::{Display, Formatter};
+
+use trust_dns_resolver::error::ResolveError;
+
+/// DNS Resolution error types
+pub enum DnsResolutionError {
+    /// Error creating the system resolver
+    CreateSystemResolver(ResolveError),
+}
```

Comment:
Use the crate `thiserror` for error creation, it abstracts the `Display` implementation